### PR TITLE
[CKImageComponent] Allow to pass in additional attributes

### PR DESCRIPTION
**Pull request by jaydee3**
In the same way CKNetworkImageComponent allows it.

### CODE REVIEW

**Comment by ocrickard** [comment]
So cool! Can we get some snapshot tests for this as well?

**Comment by jaydee3** [comment]
It's nothing new? It's an existing component.

**Comment by ocrickard** [comment]
Yeah, realize it isn't new, but we do like snapshot tests for all new view behavior in the core components. I'm cool with fixing forward if you'd prefer to do that.

**Comment by jaydee3** [comment]
Sounds great. I can add them in another PR.

File: ComponentKit/Components/CKImageComponent.h [hunk 1]
```diff
@@ -18,14 +18,16 @@ struct CKComponentSize;
...
 
 /**
- Uses a static layout with the given image size.
+ Uses a static layout with the given image size and applies additional attributes.
  */
 + (instancetype)newWithImage:(UIImage *)image
+                  attributes:(const CKViewComponentAttributeValueMap &)attributes
                         size:(const CKComponentSize &)size;
 
 @end
```
Comment (ocrickard) [inline ComponentKit/Components/CKImageComponent.h]: This is so sad that we have multiple initializers with one of our core components.

Comment (jaydee3) [inline ComponentKit/Components/CKImageComponent.h]: I'm happy to kill the first one, but it definitely adds convenience. I can split it out as a CKAutoSizeImageComponent?

Comment (ocrickard) [inline ComponentKit/Components/CKImageComponent.h]: That would be awesome. Doesn't need to be in this diff.

**Comment by ocrickard** [comment]
Looks reasonable to me. Merge into master whenever you like. Be sure to submit the internal merge soon after though.

**Comment by jaydee3** [comment]
Alright I split it out and added snapshot tests.

File: ComponentKit/Components/CKImageComponentAutoSized.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+#import <ComponentKit/CKCompositeComponent.h>
+
+struct CKComponentSize;
+
+/**
+ A component that displays an image using UIImageView.
+ */
+@interface CKImageComponentAutoSized : CKCompositeComponent
```
Comment (ocrickard) [inline ComponentKit/Components/CKImageComponentAutoSized.h]: This violates the naming convention we have. CKAutoSizedImageComponent would be appropriate.

**Comment by ocrickard** [comment]
CI failed due to snapshot tests. These need to be recorded for many different simulators sadly.

**Comment by jaydee3** [comment]
Okay it should all work now.

**Comment by ocrickard** [comment]
Bold, I like it. I'm in favor of this change, though it is a breaking API change. We've been pretty comfortable with these types of changes in the past, so long as you're cool with merging internally, I'm on board.

**Comment by jaydee3** [comment]
Let's do it!

**Pull request merged**